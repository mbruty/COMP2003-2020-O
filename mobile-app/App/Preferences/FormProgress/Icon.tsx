import React from "react";
import {
  Image,
  ImageSourcePropType,
  StyleSheet,
  Text,
  View,
} from "react-native";

interface Props {
  image: ImageSourcePropType;
  text: string;
}

export const Icon: React.FC<Props> = (props) => {
  return (
    <View style={styles.container} key={`icon ${props.text}`}>
      <Image style={styles.img} source={props.image} />
      <Text allowFontScaling={false} style={styles.txt}>{props.text}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
  },
  img: {
    height: 35,
    width: 35,
  },
  txt: {
    textAlign: "center",
  },
});
