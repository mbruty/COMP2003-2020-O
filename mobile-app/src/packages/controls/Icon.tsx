import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
interface Props {
  text: string;
  index: number;
  iconName: string;
  enabled: boolean;
}

export const Icon: React.FC<Props> = (props) => {
  const iconColour = props.enabled ? "#FD4040" : "#707070";

  return (
    <View style={styles.container} key={`icon ${props.text}`}>
      <MaterialCommunityIcons
        name={props.iconName}
        size={35}
        color={iconColour}
      />
      <Text allowFontScaling={false} style={styles.txt}>
        {props.text}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
  },
  txt: {
    textAlign: "center",
  },
});
