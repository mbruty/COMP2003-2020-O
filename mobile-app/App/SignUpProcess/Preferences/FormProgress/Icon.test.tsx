import React from "react";
import renderer from "react-test-renderer";
import { Icon } from "./Icon";

test("Renders Icon from form progress", () => {
  const circleBtnList = renderer
    .create(<Icon text="test" image={require("./account_circle_active.png")} />)
    .toJSON();
  expect(circleBtnList).toMatchSnapshot();
});
