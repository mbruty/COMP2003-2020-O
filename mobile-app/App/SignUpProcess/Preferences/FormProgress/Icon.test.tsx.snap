// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Icon from form progress 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 35,
        "width": 35,
      }
    }
  />
  <Text
    allowFontScaling={false}
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    test
  </Text>
</View>
`;
